core/src/main/resources/error/error-classes.json

```diff
@@ -3,9 +3,29 @@
     "message" : [ "Field name %s is ambiguous and has %s matching fields in the struct." ],
     "sqlState" : "42000"
   },
+  "CANNOT_GET_JDBC_TYPE" : {
+    "message" : [ "Can't get JDBC type for %s" ],
+    "sqlState" : "42000"
+  },
+  "CANNOT_READ_FOOTER_FOR_FILE" : {
+    "message" : [ "Could not read footer for file: %s" ],
+    "sqlState" : "42000"
+  },
+  "CANNOT_TRANSLATE_NON_NULL_VALUE_FOR_FIELD" : {
+    "message" : [ "Can't translate non-null value for field %s" ],
+    "sqlState" : "0A000"
+  },
   "CONCURRENT_QUERY_ERROR" : {
     "message" : [ "Another instance of this query was just started by a concurrent session." ]
   },
+  "DATA_TYPE_UNSUPPORTED_YET" : {
```

Comment:
As discussed in the [error message guidelines](https://spark.apache.org/error-message-guidelines.html), we should "Avoid time-based statements, such as promises of future support." Let's remove `yet` from the wording here.